### PR TITLE
Pin werkzeug version to fix python getting started issue

### LINKED ISSUE #3610: Python "getting started" demo does not produce output as shown on instructions
# What went wrong?
In the [Python otel getting started docs](https://opentelemetry.io/docs/instrumentation/python/getting-started/#run-the-instrumented-app), under the `run-the-instrumented-app` section, there is example output that user is to expect from the application. Reproduced from the docs:
```json
{
    "name": "/rolldice",
    "context": {
        "trace_id": "0xdb1fc322141e64eb84f5bd8a8b1c6d1f",
        "span_id": "0x5c2b0f851030d17d",
        "trace_state": "[]"
    },
    "kind": "SpanKind.SERVER",
    "parent_id": null,
    "start_time": "2023-10-10T08:14:32.630332Z",
    "end_time": "2023-10-10T08:14:32.631523Z",
    "status": {
        "status_code": "UNSET"
    },
    "attributes": {
        "http.method": "GET",
        "http.server_name": "127.0.0.1",
        "http.scheme": "http",
        "net.host.port": 8080,
        "http.target": "/rolldice?rolls=12",
        "net.peer.ip": "127.0.0.1",
        "http.user_agent": "curl/8.1.2",
        "net.peer.port": 58419,
        "http.flavor": "1.1",
        "http.route": "/rolldice",
        "http.status_code": 200,
...
```
Notice the richness of the information included: HTTP method, port, peer IP, route, status, etc. are included. However, when running the demo as-is, none of this information is included in the server logs. I've attached the full logs to this ticket (rolldice.log below) and reproduced some of the output here for ease of viewing:
```json
{
    "body": "\u001b[31m\u001b[1mWARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.\u001b[0m\n * Running on http://127.0.0.1:8080",
    "severity_number": "<SeverityNumber.INFO: 9>",
    "severity_text": "INFO",
    "attributes": {
        "otelSpanID": "0",
        "otelTraceID": "0",
        "otelTraceSampled": false,
        "otelServiceName": "dice-server"
    },
    "dropped_attributes": 0,
    "timestamp": "2023-11-27T22:09:01.452634Z",
    "trace_id": "0x00000000000000000000000000000000",
    "span_id": "0x0000000000000000",
    "trace_flags": 0,
    "resource": "BoundedAttributes({'telemetry.sdk.language': 'python', 'telemetry.sdk.name': 'opentelemetry', 'telemetry.sdk.version': '1.21.0', 'service.name': 'dice-server', 'telemetry.auto.version': '0.42b0'}, maxlen=None)"
}
{
    "body": "\u001b[33mPress CTRL+C to quit\u001b[0m",
    "severity_number": "<SeverityNumber.INFO: 9>",
    "severity_text": "INFO",
    "attributes": {
        "otelSpanID": "0",
        "otelTraceID": "0",
        "otelTraceSampled": false,
        "otelServiceName": "dice-server"
    },
    "dropped_attributes": 0,
    "timestamp": "2023-11-27T22:09:01.453917Z",
    "trace_id": "0x00000000000000000000000000000000",
    "span_id": "0x0000000000000000",
    "trace_flags": 0,
    "resource": "BoundedAttributes({'telemetry.sdk.language': 'python', 'telemetry.sdk.name': 'opentelemetry', 'telemetry.sdk.version': '1.21.0', 'service.name': 'dice-server', 'telemetry.auto.version': '0.42b0'}, maxlen=None)"
}
{
    "body": "me is rolling the dice: 2",
    "severity_number": "<SeverityNumber.WARN: 13>",
    "severity_text": "WARNING",
    "attributes": {
        "otelSpanID": "0",
        "otelTraceID": "0",
        "otelTraceSampled": false,
        "otelServiceName": "dice-server"
    },
    "dropped_attributes": 0,
    "timestamp": "2023-11-27T22:09:22.982898Z",
    "trace_id": "0x00000000000000000000000000000000",
    "span_id": "0x0000000000000000",
    "trace_flags": 0,
    "resource": "BoundedAttributes({'telemetry.sdk.language': 'python', 'telemetry.sdk.name': 'opentelemetry', 'telemetry.sdk.version': '1.21.0', 'service.name': 'dice-server', 'telemetry.auto.version': '0.42b0'}, maxlen=None)"
}
```

# Why care?
Notice that it extracts none of the richness (HTTP method etc.) shown in the docs. Were that the only issue, maybe that's ok, but also notice that it zeros out all the trace and span IDs. That seems like more of an issue.

Moreover, none of the histogram information e.g.
```json

                    "bucket_counts": [0, 1, 0, 0, 0, 0, 0, 0, 0, 0, 0],
                    "count": 1,
                    "explicit_bounds": [
                      0, 5, 10, 25, 50, 75, 100, 250, 500, 1000
                    ],
                    "max": 1,
                    "min": 1,
                    "start_time_unix_nano": 1666077040063027610,
                    "sum": 1,
                    "time_unix_nano": 1666077098181107419
```
as shown on the docs was produced after running the demo, curling upwards of 30 times, and waiting upwards of 60 seconds. `cat rolldice.log | grep bucket` to confirm.

[rolldice.log](https://github.com/open-telemetry/opentelemetry.io/files/13481066/rolldice.log)

### PR DESCRIPTION
**Pull request by svrnm**
fixes #3610

Note that this is a hot fix, we should do some additional improvements to the python getting started in the future to avoid those kinds of issues from happening (i.e. pin the versions of all dependencies)

### CODE REVIEW

**Comment by ekquasar** [comment]
I cannot review (I don't think) but I can validate that pinning `werkzeug<3` fixes the issue.
When I install I get `2.3.8` not `2.3.7`, maybe pin to `<3` to indicate that the major version increment in werkzeug introduced the breaking change.

File: content/en/docs/instrumentation/python/getting-started.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ source ./venv/bin/activate
 Now install Flask:
 
 ```shell
-pip install 'flask<3'
+pip install 'flask<3' 'werkzeug==2.3.7'
```
Comment (svrnm) [inline content/en/docs/instrumentation/python/getting-started.md:46]:
```suggestion
pip install 'flask<3' 'werkzeug<3'
```

**Comment by svrnm** [comment]
> I cannot review (I don't think) but I can validate that pinning `werkzeug<3` fixes the issue. 

You can always review PRs on github, and they are helpful to drive the PR, but your review will not satisfy the review requirements.

> When I install I get `2.3.8` not `2.3.7`, maybe pin to `<3` to indicate that the major version increment in werkzeug introduced the breaking change.

I updated it to `<3` thanks

**Review by chalin — APPROVED** [review]
RSLGTM

**Review by cartermp — APPROVED** [review]

**Pull request merged**